docs/DeveloperGuide.md

```diff
@@ -1,38 +1,323 @@
+
+`ExpenseFilter` is a prerequisite to implementing other features, e.g. Expense statistics and visualization.
+
+<u>Implementation Details</u>
+
+`ExpenseFilter` filters expenses by 3 criteria: Category, From (date), To (date).
+Each criterion is taken care of by a helper method.
+Since these filter fields are optional, if they are `null`, helper methods will evaluate to `true`.
+
+Given a list of expenses, `ExpenseFilter` iterates through each expense, applying three criteria checks on them; and 
+it would add the expense to a new `ArrayList` if all three checks are satisfied.
+
+The `ArrayList` is then returned to the caller.
+
+### Date and Time Handling
+
+The `DateUtils` class provides utility methods to handle date formatting, validation and conversion. 
+
+The `DateUtils` class has no notable methods.
+
+<u>Implementation Details</u>
+
+The `DateUtils` class is implemented as a Singleton as its methods are common to all other classes that require it.
+
+### Recurring Expense and Recurring Expense List
+
+<u>Overview</u>
+
+The `RecurringExpense` class extends from the `Expense` class and it represents an indivual recurring expense with a price, description, category, last date added and a frequency.
+
+The `RecurringExpenseList` class extends from the `ExpenseList` class and it manages a collection of `RecurringExpense` objects.
+It allows for addition, editing and deletion of expenses.
+
+<u>Methods</u>
+
+The `RecurringExpense` class has no notable methods.
+
+The `RecurringExpenseList` class has the following key methods:
+
+|          Method          |                Description                 |
+|:------------------------:|:------------------------------------------:|
+|  `addRecurringExpense`   |    Adds a recurring expense to the list    |
+| `deleteRecurringExpense` | Removes a recurring expense from the list  |
+|  `editRecurringExpense`  |   Edits a recurring expense in the list    |
+| `loadFromCsv` | Adds the appropriate amount of `Expense` objects with the correct date to the `ExpenseList`|
+
+
+<u>Design Considerations</u>
+
+Since the programme does not have an auto-save function upon closing the programme or auto-load when starting the programme, it is up to the user to save their work and to load it again.
+
+To minimise the amount of checks that need to be done, the recurring expenses are only added after the user calls the `load` command.
+
+<u>Implementation Details</u>
+
+Below is the UML class diagram for `RecurringExpense` and `RecurringExpenseList`:
+
+![RecurringExpenseAndRecurringExpenseList.png](diagrams%2Fimages%2FRecurringExpenseAndRecurringExpenseList.png "UML Class Diagram for RecurringExpense and RecurringExpenseList class")
+
+Below is the sequence diagram for when the user calls the `load` command.
+
+![RecurringExpenseLoadFromCsvSequence.png](diagrams%2Fimages%2FRecurringExpenseLoadFromCsvSequence.png "UML Sequence Diagram for calling load command")
+
```

Comment:
When instantiating new objects for File, FileReader and CSVReader, should the arrow point to the side of the object box, with an activation box coming right out of the object box to represent the constructor? And maybe you could perhaps not include the "new" keyword in the arrow? Can refer to https://nus-cs2113-ay2425s1.github.io/website/schedule/week10/topics.html#W10-1 for the proper formatting for object creation in sequence diagrams.